### ISSUE TITLE
Find unused exception classes and remove

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#701]
The ga4gh/exceptions.py module is the central repository for exceptions, which makes importing them and keeping track of the inheritance hierarchy easy. However, it can be hard to keep track of which ones are being used. 

Find the exceptions which are currently not being used in the code base and delete them. 
For example,

```bash
$ grep -IR BadReadsSearchRequestBothRefs ga4gh
ga4gh/exceptions.py:class BadReadsSearchRequestBothRefs(BadRequestException):
```

Since the only match we have is the exceptions.py file, this exception isn't being used anywhere and can be removed.

Try using the [vulture](https://pypi.python.org/pypi/vulture) package to find other unused exceptions. The output of vulture contains lots of false positives, so beware. However, it does find some other unused exceptions.

Can we build an automated way we can include in our Travis setup to detect unused exceptions using vulture?

**Comment by denis-yuen** [comment]
python neophyte here, will take a look at this

**Comment by dcolligan** [comment]
Fixed in #729
